packages/client/src/generation/generateClient.ts

```diff
@@ -539,5 +545,18 @@ async function copyRuntimeFiles({ from, to, runtimeName, sourceMaps }: CopyRunti
     files.push(...files.filter((file) => file.endsWith('.js')).map((file) => `${file}.map`))
   }
 
-  await Promise.all(files.map((file) => fs.promises.copyFile(path.join(from, file), path.join(to, file))))
+  await Promise.all(files.map((file) => fs.copyFile(path.join(from, file), path.join(to, file))))
+}
+
+/**
+ * Attempts to delete the output directory.
+ * @param finalOutputDir
+ */
+async function deleteOutputDir(finalOutputDir: string) {
+  try {
+    // we want to make sure that if we delete, we delete the right directory
+    if (require(`${finalOutputDir}/package.json`).name === GENERATED_PACKAGE_NAME) {
+      await fs.rmdir(finalOutputDir, { recursive: true })
+    }
```

Comment:
`--no-engine` will delete the output directory of the client. Do you want me to perhaps delete each individual file instead?

Reply:
I was also wondering about this, as it makes the behaviour a bit different, but I'm not sure about whether there are any real implications.

One thing that comes to my mind, can it cause any issues for `prisma generate --watch`?

Reply:
Are we testing watch? If not I can test it by hand, report, and open an issue to get it tested.

Reply:
I checked and currently `--no-engine` is not implemented for `--watch`.